server/src/main/java/io/crate/execution/dml/ObjectIndexer.java

```diff
@@ -159,15 +159,33 @@ public void collectSchemaUpdates(@Nullable Map<String, Object> value,
 
     @Override
     public void updateTargets(Function<ColumnIdent, Reference> getRef) {
+        ColumnIdent objectColumn = ref.column();
+        Reference updatedSelf = getRef.apply(objectColumn);
+        ObjectType objectType = (ObjectType) ArrayType.unnest(updatedSelf.valueType());
```

Comment:
Why do we need unnesting from ArrayType? Isn't `ObjectIndexer` suposed to be created from a `Reference` with type = OBJECT?